x-pack/solutions/observability/plugins/infra/server/features.ts

```diff
@@ -32,69 +34,87 @@ const metricRuleTypes = [
   ML_ANOMALY_DETECTION_RULE_TYPE_ID,
 ];
 
-const metricAlertingFeatures = metricRuleTypes.map((ruleTypeId) => ({
-  ruleTypeId,
-  consumers: [METRICS_FEATURE_ID, ALERTING_FEATURE_ID, ...DEPRECATED_ALERTING_CONSUMERS],
-}));
+export const getMetricsAlertingFeatures = (
+  isServerlessEnv: boolean = false
+): KibanaFeatureConfig => {
+  const metricAlertingFeatures = metricRuleTypes.map((ruleTypeId) => {
+    const consumers = [
+      METRICS_FEATURE_ID,
+      ALERTING_FEATURE_ID,
+      ...DEPRECATED_ALERTING_CONSUMERS,
+      // Add STACK_ALERTS only if in serverless and only for observability threshold and es query rules
+      ...(isServerlessEnv &&
+      (ruleTypeId === OBSERVABILITY_THRESHOLD_RULE_TYPE_ID || ruleTypeId === ES_QUERY_ID)
```

Comment:
I honestly don't think this conditional is strickly necessary. Yes, in our testing we may have only noted that stackAlerts is applied to threshold rule and es query rule, but in our general testing one thing that I've internalized is that the consumer is often unpredictable. I don't think we're hurting anything by indiscriminately allowing the stack alerts consumer is serverless. And we reduce the cognitive load on future engineers.